modules/data.atmosphere/R/download.NEONmet.R

```diff
@@ -0,0 +1,486 @@
+          csvData <- read.csv(url30,stringsAsFactors=FALSE,header=TRUE)
+          #Retreive time dimension and figure out where in array to put it
+          csvDateTime <- as.POSIXct(gsub("T"," ",csvData$startDateTime),tz="UTC")
+          arrLoc <- floor(as.numeric(difftime(csvDateTime,datetime[1],tz="UTC",units="hours"))*2)+1
+          csvVar <- csvData$RHMean
+          csvQF <- csvData$RHFinalQF
+          csvVar[which(csvQF!=0)]=NA
+          ncdata[arrLoc] <- udunits2::ud.convert(csvVar,"%", "%")
+        }
+      }
+      ncdf4::ncvar_put(nc, varid = RH.var, vals = ncdata)
+    }
+    
+    # DP1.00023 SW/LW or NEON.DP1.00022 SW (Possible future: DP1.00014 for Direct/Diffuse SW)
+    SWLoc = grep("DP1.00022",availProducts)
+    SWLWLoc = grep("DP1.00023",availProducts)
+    if (length(SWLoc)!=0) { 
+      SWDates = unlist(availDates[SWLoc[1]])
+      SWGoodDates = which((SWDates >= startMon) & (SWDates <= endMon))
+      nSW = length(SWGoodDates)
+    } else { nSW = 0 }
+    if (length(SWLWLoc)!=0) { 
+      SWLWDates = unlist(availDates[SWLWLoc[1]])
+      SWLWGoodDates = which((SWLWDates >= startMon) & (SWDates <= endMon))
+      nSWLW = length(SWLWGoodDates)
+    } else { nSWLW = 0 }
+
+    if (nSWLW!=0) {
+      SW.var <- ncvar_def(name = "surface_downwelling_shortwave_flux_in_air", 
+                          units = "W m-2", 
+                          dim = xytdim)
+      nc <- ncdf4::ncvar_add(nc = nc, v = SW.var, verbose = verbose)
+      LW.var <- ncvar_def(name = "surface_downwelling_longwave_flux_in_air",
+                          units = "W m-2", 
+                          dim = xytdim)
+      nc <- ncvar_add(nc = nc, v = LW.var, verbose = verbose)
+      ncdata <- rep(FillValue,length(datetime))
+      ncdata2 <- rep(FillValue,length(datetime))
+      for (mon in SWLWDates[SWLWGoodDates]) {
+        neonData <- nneo::nneo_data(product_code = availProducts[SWLWLoc[1]], site_code = site, year_month = mon)
+        if (length(neonData$data$urls)>0) {
+           #Extract and read 30 minute data from the highest vertical level among files returned
+          url30 <- tail(sort(neonData$data$urls[grep("*.00000.000.*_30min",neonData$data$urls)]),1)
+          csvData <- read.csv(url30,stringsAsFactors=FALSE,header=TRUE)
+          #Retreive time dimension and figure out where in array to put it
+          csvDateTime <- as.POSIXct(gsub("T"," ",csvData$startDateTime),tz="UTC")
+          arrLoc <- floor(as.numeric(difftime(csvDateTime,datetime[1],tz="UTC",units="hours"))*2)+1
+          csvVar <- csvData$inSWMean
+          csvQF <- csvData$inSWFinalQF
+          csvVar[which(csvQF!=0)]=NA  
+          ncdata[arrLoc] <- udunits2::ud.convert(csvVar,"W m-2", "W m-2")
+          csvVar <- csvData$inLWMean
+          csvQF <- csvData$inLWFinalQF
+          csvVar[which(csvQF!=0)]=NA  
+          ncdata2[arrLoc] <- udunits2::ud.convert(csvVar,"W m-2", "W m-2")
```

Comment:
These are unnecessary unit conversions. I'm guessing they're here because you copy-pasted a code block and revised small pieces of it, which is an excellent indication that a function may be a good idea here.